gui/dewar_tree.py

```diff
@@ -272,19 +272,19 @@ def add_samples_to_puck_tree(
             self.parent.row_clicked(current_index)
 
     def is_proposal_member(self, proposal_id) -> bool:
-        # Check if the user running LSDC is part of the sample's proposal
-        if proposal_id not in self.proposal_membership:
-            r = requests.get(f"{os.environ['NSLS2_API_URL']}/proposal/{proposal_id}")
+        # Check if the user running LSDC is part of the sample's proposal 
+        # If the proposal_membership set is empty get data from API (Use prod API)
+        # This way API is only polled once when GUI starts
+        if not self.proposal_membership:
+            r = requests.get(f"{os.environ['NSLS2_API_URL']}/v1/data-session/{getpass.getuser()}")
             r.raise_for_status()
             response = r.json()
-            if "users" in response and getpass.getuser() in [
-                user["username"] for user in response["users"] if "username" in user
-            ]:
-                self.proposal_membership[proposal_id] = True
-            else:
-                logger.info(f"Users not found in response: {response}")
-                self.proposal_membership[proposal_id] = False
-        return self.proposal_membership[proposal_id]
+            for session in response.get("data_sessions", []):
+                # Assuming data_sessions has values of the form "pass-123456"
+                proposal = session.split("-")[1]
+                self.proposal_membership.add(proposal)
```

Comment:
actually, this won't take care of case 2 above, since you only add proposals, never remove. so would require clearing all proposals and re-creating the set.